File: golang/191_number_of_1_bits.go
package main

import (
	"fmt"
)

func hammingWeight(num uint32) int {
	var count int
	for num > 0 {
		// fmt.Println("it:: ", num)
		count += int(num & 1)
		num >>= 1
	}
	return count
}

func main() {
	var i uint32
	fmt.Scanf("%d", &i)
	fmt.Println(i)
	fmt.Println(hammingWeight(i))
}

File: python/191_number_of_1_bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n:
            count += (n % 2)
            n = n >> 1
        return count

sol = Solution()
inp = int(input())
print("inp:: ", inp)
print(sol.hammingWeight(inp))